docs/reference/transforms/from_deps.rst

```diff
@@ -111,20 +111,43 @@ groups:
 Then the ``notify`` task will be duplicated into three, one for each group. The
 notify tasks will depend on each task in its associated group.
 
+You may also provide a special value of ``all`` to the ``group-by`` function.
+Using ``all`` will *always* result in one task being generated, with all tasks
+from the included kinds to be set as dependencies. It is usually useful to also
+set ``unique-kinds`` to ``False`` when using ``all``.
+
+If we alter the ``kind`` definition from above as follows:
+
+.. code-block:: yaml
+
+   kind-dependencies:
+     - build
+     - signing
+     - publish
+
+   tasks:
+     notify:
+       from-deps:
+         group-by: all
+
+We would end up with a single ``notify`` task that depends on all tasks from
+the ``build``, ``signing``, and ``publish`` kinds.
+
+
```

Comment:
Thanks for the docs! I guess we should have a dedicated reference section for all of the built-in `group-by` functions, but that can be a follow-up and this is great for now.